### ISSUE TITLE
Making many fault sectors can avoid dropping power

### ISSUE THREAD

**Issue by coder-lb** [filecoin-project/specs-actors#569]
### Problems
We did some tests several days before, and it didn't submit any WindowPost. In theory, it should drop all the power, but it didn't happened.  We got the error like this:
```
WARN    vm      vm/runtime.go:367       vmctx send failed: to: t0xxxx, method: 12: ret: [], err: failed to marshal input parameters: Slice value in field t.Weights was too long (RetCode=4)
it means the fault sectors more than 8192.  it will broken the dealine index in 36.
```
Some other miners also got the similar errors.

The method 12  is OnDeferredCronEvent.

The corresponding code causes OnDeferredCronEvent not to run as expected:

```
func (t *OnFaultBeginParams) MarshalCBOR(w io.Writer) error {
	if t == nil {
		_, err := w.Write(cbg.CborNull)
		return err
	}
	if _, err := w.Write([]byte{129}); err != nil {
		return err
	}

	// t.Weights ([]power.SectorStorageWeightDesc) (slice)
	if len(t.Weights) > cbg.MaxLength {
		return xerrors.Errorf("Slice value in field t.Weights was too long")
	}

	if _, err := w.Write(cbg.CborEncodeMajorType(cbg.MajArray, uint64(len(t.Weights)))); err != nil {
		return err
	}
	for _, v := range t.Weights {
		if err := v.MarshalCBOR(w); err != nil {
			return err
		}
	}
	return nil
}
```

There are some improvements in latest code. For example, it changes the method of calculating power. It should avoid current issue. But it's also dangerous if something is larger than 8192. More improvement are needed.  The function handleProvingPeriod() is very important, and it is easy to construct attacks.

**Comment by hyunmoon** [comment]
Same issue here. [https://filecoinproject.slack.com/archives/CEGN061C5/p1593565514264800](url)

**Comment by anorth** [comment]
This is already fixed on master by #488 , but there are similar issues we're tracking like #391
